### PR TITLE
overlord/ifacestate: report bad plug/slots with warnings on snap install

### PR DESCRIPTION
**Pull request by stolowski**
This is to address issues like https://bugs.launchpad.net/snapd/+bug/1676492 where snap install silently ignores bad plugs/slots (they are reported in task log only), so user has no idea something was wrong with the snap. Gustavo's suggestion from the forum topic (linked in the bug) was to address this once we have proper warnings infrastructure (which we now have), so this PR does that.

### CODE REVIEW

**Comment by mvo5** [comment]
The failure in:
```
019-11-16 06:40:09 Error executing google:ubuntu-core-16-64:tests/main/bad-interfaces-warn : 
-----
+ echo 'Installing a snap with invalid interfaces issues a warning'
Installing a snap with invalid interfaces issues a warning
+ MATCH 'WARNING: There is 1 new warning. See '\''snap warnings'\'''
+ snap install --dangerous test-snap_1_all.snap
+ echo 'And the warning contains the list of bad interfaces'
And the warning contains the list of bad interfaces
+ MATCH '.*snap "test-snap" has bad plugs or slots: badinterface1 \(unknown interface'
+ snap warnings
+ MATCH 'badinterface2 \(unknown interface "foo"\)'
+ snap warnings
+ echo 'And the correct interface is connected'
And the correct interface is connected
+ MATCH 'home *test-snap:home *:home'
+ snap connections test-snap
grep error: pattern not found, got:
Interface  Plug            Slot  Notes
home       test-snap:home  -     -
-----
```
looks real

**Comment by stolowski** [comment]
> The failure in:
> 
> ```
> 019-11-16 06:40:09 Error executing google:ubuntu-core-16-64:tests/main/bad-interfaces-warn : 
> ```
> 
> looks real

Right, that was a test issue, I forgot that home is not connected on core by default. Should be fixed now.

**Comment by pedronis** [comment]
it would be great to work to land #7320  as well/first

File: overlord/ifacestate/ifacestate_test.go [hunk 1]
```diff
@@ -2094,6 +2094,44 @@ func (s *interfaceManagerSuite) TestDoSetupSnapSecurityHonorsUndesiredFlag(c *C)
 	c.Assert(ifaces.Connections, HasLen, 0)
 }
 
+func (s *interfaceManagerSuite) TestBadInterfacesWarning(c *C) {
+	restoreSanitize := snap.MockSanitizePlugsSlots(func(inf *snap.Info) {
+		inf.BadInterfaces["a"] = "b"
```
Comment (pedronis) [inline overlord/ifacestate/ifacestate_test.go]: for clarity:

s/a/plug-name/
s/b/reason-for-bad/

**Pull request merged**